File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.Duration;


class Gigasecond {
    
    public final Duration GIGASECOND = Duration.ofSeconds(1_000_000_000);
    LocalDateTime time;

    Gigasecond(LocalDate moment) {
        this(moment.atStartOfDay());
    }

    Gigasecond(LocalDateTime moment) {
        time = moment.plus(GIGASECOND);
    }

    LocalDateTime getDateTime() {
        return time;    
    }

}

File: kotlin/gigasecond/bin/main/Gigasecond.kt
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.temporal.ChronoUnit;

class Gigasecond(localDateTime: LocalDateTime) {
    constructor(localDate: LocalDate): this(localDate.atStartOfDay())

    val date = localDateTime.plus(1000000000, ChronoUnit.SECONDS)
}